tools/explorer/tt_adapter/src/tt_adapter/main.py

```diff
@@ -70,9 +74,23 @@ def execute(
             memory_layout_analysis_enabled = False
             memory_layout_analysis_policy = None
 
-        ttnn_ir = self.model_runner.run(
+        self.model_runner.run(
             model_path, memory_layout_analysis_enabled, memory_layout_analysis_policy
         )
 
         # TODO(odjuricic, #933) Parse TTNN IR and return the post optimized graph.
         return {"graphs": []}
+
+    def status_check(self, model_path: str, settings: Dict) -> bool:
+        done = not self.model_runner.is_busy()
+        error = self.model_runner.get_error()
+        progress = self.model_runner.get_progress()
+
+        return utils.to_adapter_format(
+            {
+                "isDone": done,
+                "progress": progress,
+                "total": 100,
+                "error": error,
```

Comment:
Would it be interesting to add properties to send logs to the UI, like, actual text that gives a better perspective of what is going on?

Reply:
Adding that today. Left a comment on your PR to discuss these properties.